src/coreclr/vm/methodtable.cpp

```diff
@@ -4467,24 +4331,91 @@ void MethodTable::CheckRunClassInitThrowing()
     // To find GC hole easier...
     TRIGGERSGC();
 
-    if (IsClassPreInited())
-        return;
-
     // Don't initialize shared generic instantiations (e.g. MyClass<__Canon>)
     if (IsSharedByGenericInstantiations())
         return;
 
-    DomainLocalModule *pLocalModule = GetDomainLocalModule();
-    _ASSERTE(pLocalModule);
+    EnsureStaticDataAllocated();
+    DoRunClassInitThrowing();
+}
+
+void MethodTable::EnsureStaticDataAllocated()
+{
+    WRAPPER_NO_CONTRACT;
```

Comment:
Why no contract?

Reply:
Because I'm lazy? Yep, that's it. In any case, pulling the contract from the Allocate functions is reasonable, so I'll do that.